frontend/heir/pipeline.py

```diff
@@ -231,16 +251,22 @@ def compile(
       backend = OpenFHEBackend(openfhe_config.from_os_env())
 
   if debug and heir_opt_options is not None:
-    print(f"HEIR Debug: Overriding scheme with options {heir_opt_options}")
+    DebugMessage(f"Overriding scheme with options {heir_opt_options}")
 
   def decorator(func):
-    return run_pipeline(
-        func,
-        heir_opt_options=heir_opt_options
-        or ["--canonicalize", f"--mlir-to-{scheme}"],
-        backend=backend or CleartextBackend(),
-        heir_config=config or heir_cli_config.from_os_env(),
-        debug=debug or False,
-    )
+    try:
+      return run_pipeline(
+          func,
+          heir_opt_options=heir_opt_options
+          or ["--canonicalize", f"--mlir-to-{scheme}"],
+          backend=backend or CleartextBackend(),
+          heir_config=config or heir_cli_config.from_os_env(),
```

Comment:
```suggestion
          backend=backend,
          heir_config=config,
```

These defaults are now unreachable